pkg/apis/v1/dnsrecord_types.go

```diff
@@ -105,15 +105,22 @@ func (e *Endpoint) String() string {
 
 // DNSRecordSpec defines the desired state of DNSRecord
 type DNSRecordSpec struct {
+	// +kubebuilder:validation:Required
+	// +required
+	ManagedZoneRef *ManagedZoneReference `json:"managedZone,omitempty"`
 	// +kubebuilder:validation:MinItems=1
 	// +optional
 	Endpoints []*Endpoint `json:"endpoints"`
 }
 
 // DNSRecordStatus defines the observed state of DNSRecord
 type DNSRecordStatus struct {
-	// zones are the status of the record in each zone.
-	Zones []DNSZoneStatus `json:"zones,omitempty"`
+
+	// conditions are any conditions associated with the record in the managed zone.
+	//
+	// If publishing the record fails, the "Failed" condition will be set with a
+	// reason and message describing the cause of the failure.
+	Conditions []DNSRecordCondition `json:"conditions,omitempty"`
```

Comment:
Any reason not to use `Conditions []metav1.Condition` here

Reply:
Updated to use metav1.Condition